reactor-core/src/main/java/reactor/core/publisher/FluxWindowWhen.java

```diff
@@ -235,15 +237,16 @@ void drainLoop() {
 						dispose();
 						Throwable e = error;
 						if (e != null) {
-							actual.onError(e);
```

Comment:
Instructions here and in in `onComplete()` got swapped. Previously, the next line(s) would have no effect, as the call to `actual` cancelled the window subscriptions, causing the signals to be lost.